src/utilities/functions.ts

```diff
@@ -62,39 +68,77 @@ export function gitFix(path:string) : string {
 	return path.replace(c.gitExtensionRegExp, '');
 }
 
+export function executeShellCommand(command:string, cwd:vscode.Uri, errorMessage:string) : Answer {
+	let out = {stdout:'', stderr:''};
+	cp.exec(command, {cwd:cwd.fsPath}, (_, stdout, stderr) => {
+		out = {stdout:stdout, stderr:stderr};
+	});
+	if (out.stderr) {
+		void vscode.window.showErrorMessage(`${errorMessage}: ${out.stderr}`);
+	}
+	return out;
+}
+
+export async function executeShellCommandWithProgressBar(command:string, cwd:vscode.Uri, progressTitle:string, errorMessage:string) : Promise<Answer> {
+	// run a shell command and show a moving progress bar in the mean time
+	let out:Answer = {stdout:'', stderr:''};
+	await vscode.window.withProgress(
+		{location: vscode.ProgressLocation.Notification, cancellable: false, title: progressTitle}, 
+		async (progress) => {
+			// create progress bar at 0%
+			progress.report({  increment: 0 });
+			// pointer with 'done' status which will be updated by the command once finished, 
+			// and monitored by the progress bar to close once updated 
+			const progressDetails = { isDone: false };
+			// execute shell command 
+			cp.exec(command, {cwd: cwd.fsPath}, (_, stdout, stderr) => {
+				// once finished: update 'done' status, close progress bar
+				out = {stdout:stdout, stderr:stderr};
+				progress.report({ increment: 100 });
+				progressDetails.isDone = true;
+				return;
+			});
+			// update progress bar while not done
+			await fakeProgressUpdate(progress, progressDetails);			
+		}
+	);
+	if (out.stderr) {
+		void vscode.window.showErrorMessage(`${errorMessage}: ${out.stderr}`);
+	}
+	return out;
+}
+
 export async function delay(ms: number) {
     return new Promise( resolve => setTimeout(resolve, ms) );
 }
 
-export async function fakeProgressUpdate(progressParameter:Progress, progress: { isDone:boolean }) : Promise<void> {
+export async function fakeProgressUpdate(progressBar:ProgressBar, executionStatus: { isDone:boolean }) : Promise<void> {
```

Comment:
Is this not an internal function? If so you would not need to export it.

Reply:
fixed for all internally-used functions